Updated Pipfile to allow latest black release

bryant-finney opened the pull request:
# Summary

- changed `black==19.3b0` to `black>=19.3b0`
- formatted source with `black==19.10b0` 
  - trailing commas were added to single-argument multi-line function definitions
- handful of dependency updates in `Pipfile.lock`

Signed-off-by: Bryant Finney <bryant@outdoorlinkinc.com>

Madoshakalaka commented:
looks good! have to say I forget to remove deploy stage for pull requests (that's why you travis checks failed at last). Anyways good job!

Madoshakalaka commented:
Hi I had a closer look and there may be some thing you overlook:

1. pipenv-setup was already using 19.3b0 not 13.3b0 (typo of yours?)
1. black is in beta and versioning with inequalities does not make sense. One can only use "==" in pipfile. With "black>=19.3b0" in Pipfile, one will get error while `pipenv install --dev`
```
ERROR: ERROR: Could not find a version that matches black>=19.3b0                                                                   Skipped pre-versions: 18.3a0, 18.3a0, 18.3a1, 18.3a1, 18.3a2, 18.3a2, 18.3a3, 18.3a3, 18.3a4, 18.3a4, 18.4a0, 18.4a0, 18.4a1, 18.4a1, 18.4a2, 18.4a2, 18.4a3, 18.4a3, 18.4a4, 18.4a4, 18.5b0, 18.5b0, 18.5b1, 18.5b1, 18.6b0, 18.6b0, 18.6b1, 18.6b1, 18.6b2, 18.6b2, 18.6b3, 18.6b3, 18.6b4, 18.6b4, 18.9b0, 18.9b0, 19.3b0, 19.3b0, 19.10b0, 19.10b0                                                      There are incompatible versions in the resolved dependencies.
```

Madoshakalaka commented:
@bryant-finney just an update, I've bumped black to 19.10b in masters and reformatted files.

Just published 2.2.0 with some new functionalities

bryant-finney commented:
> Hi I had a closer look and there you may overlooked this:
> 
> black is in beta and versioning with inequalities does not make sense. One can only use "==" in pipfile. With "black>=19.3b0" in Pipfile, one will get error while `pipenv install --dev`
> 
> ```
> ERROR: ERROR: Could not find a version that matches black>=19.3b0
> 
> Skipped pre-versions: 18.3a0, 18.3a0, 18.3a1, 18.3a1, 18.3a2, 18.3a2, 18.3a3, 18.3a3, 18.3a4, 18.3a4, 18.4a0, 18.4a0, 18.4a1, 18.4a1, 18.4a2, 18.4a2, 18.4a3, 18.4a3, 18.4a4, 18.4a4, 18.5b0, 18.5b0, 18.5b1, 18.5b1, 18.6b0, 18.6b0, 18.6b1, 18.6b1, 18.6b2, 18.6b2, 18.6b3, 18.6b3, 18.6b4, 18.6b4, 18.9b0, 18.9b0, 19.3b0, 19.3b0, 19.10b0, 19.10b0                                                      
> 
> There are incompatible versions in the resolved dependencies.
> ```

Ah yeah, you're right, I hadn't included the `--pre` flag with `pipenv install --dev`. Now that `pytest-black` is gone...

```bash
pipenv install --dev --pre
Installing dependencies from Pipfile.lock (1200d4)…
Ignoring autopep8: markers 'python_version < "3.6"' don't match your environment
  🐍   ▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉ 66/66 — 00
```

Rebased PR with an update to Pipfile that sets `allow_prereleases = true`.  Also added a couple of `.gitignore` items 🙂

Also TIL about using markers to limit dependencies to specific python versions. Nice!

Madoshakalaka commented:
never knew there's `allow_prereleases = True` magic, good job!
